File: src/backend/app/models/project.py
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base

class Project(Base):
    __tablename__ = "projects"

    id = Column(Integer, primary_key=True, index=True)
    buyer_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    title = Column(String, nullable=False, index=True)
    description = Column(String, index=True)

    buyer = relationship("User", back_populates="projects")
    tasks = relationship("Task", back_populates="project")

File: src/client/lib/shared/models/project.dart
class Project {
  final int id;
  final int buyerId;
  final String title;
  final String description;

  Project({required this.id, required this.buyerId, required this.title, required this.description});

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json['id'],
      buyerId: json['buyer_id'],
      title: json['title'],
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
    };
  }
}